File: app/src/release/java/org/mifos/mobilebanking/api/BaseURL.java
package org.mifos.mobilebanking.api;

/**
 * @author Vishwajeet
 * @since 09/06/16
 */

public class BaseURL {

    public static final String API_ENDPOINT = "demo.openmf.org";
    public static final String API_PATH = "/fineract-provider/api/v1/self/";
    public static final String PROTOCOL_HTTPS = "https://";

    private String url;

    public String getUrl() {
        if (url == null) {
            return PROTOCOL_HTTPS + API_ENDPOINT + API_PATH;
        }
        return url;
    }

    public String getDefaultBaseUrl() {
        return PROTOCOL_HTTPS + API_ENDPOINT;
    }

    public String getUrl(String endpoint) {
        return endpoint + API_PATH;
    }
}

File: app/src/release/java/org/mifos/mobile/api/BaseURL.kt
package org.mifos.mobile.api

import org.mifos.mobile.api.BaseURL

/**
 * @author Vishwajeet
 * @since 09/06/16
 */
class BaseURL {
    val url: String? = null
        get() = field
                ?: PROTOCOL_HTTPS + API_ENDPOINT + API_PATH
    val defaultBaseUrl: String
        get() = PROTOCOL_HTTPS + API_ENDPOINT

    fun getUrl(endpoint: String): String {
        return endpoint + API_PATH
    }

    companion object {
        const val API_ENDPOINT = "mobile.openmf.io"
        const val API_PATH = "/fineract-provider/api/v1/self/"
        const val PROTOCOL_HTTPS = "https://"
    }
}